core/src/main/scala/package.scala

```diff
@@ -75,22 +81,30 @@ package object nelson {
 
   implicit class BedazzledOpt[A](in: Option[A]){
 
-    private def fail[B](err: NelsonError): Task[B] =
-      Task.fail(err)
+    private def fail[B](err: NelsonError): IO[B] =
+      IO.raiseError(err)
 
     def nfold[B](e: NelsonError)(f: A => B): NelsonK[B] =
       tfold(e)(f).liftKleisli
 
-    def tfold[B](e: NelsonError)(f: A => B): Task[B] =
-      in.fold(fail[B](e))(a => Task.delay(f(a)))
+    def tfold[B](e: NelsonError)(f: A => B): IO[B] =
+      in.fold(fail[B](e))(a => IO(f(a)))
   }
 
-  implicit class BedazzledTask[A](in: Task[A]){
-    def retryExponentially(seed: Duration = 15.seconds, limit: Int = 5): Task[A] = {
-      val periods = List.fill(limit)(seed).zipWithIndex.map {
-        case (d,i) => d * i.toDouble
+  implicit class BedazzledIO[A](in: IO[A]){
+    def retryExponentially(seed: FiniteDuration = 15.seconds, limit: Int = 5)(scheduler: ScheduledExecutorService, ec: ExecutionContext): IO[A] = {
```

Comment:
Needs a look

Reply:
I'm not familiar with this `retry` on fs2, but sees legit.